stingray/gti.py

```diff
@@ -194,8 +194,8 @@ def cross_two_gtis(gti0, gti1):
     cross_gtis : From multiple GTI lists, extract common intervals *EXACTLY*
 
     """
-    gti0 = np.array(gti0, dtype=np.longdouble)
-    gti1 = np.array(gti1, dtype=np.longdouble)
+    gti0 = np.asarray(gti0)
```

Comment:
I thought it over, and my take is that it would be better to give the possibility to give integers or whatever as GTIs, and respect the choice. For example, it's simpler for tests, where `cross_two_gtis(g0, g1) == [[1, 2]]` would not always work if we force it to be a float comparison.